CHANGELOG.md

```diff
@@ -1,5 +1,43 @@
 # Changelog
 
+## 0.4.27
+
+This release includes support for the `[dependency-groups]` table as recently standardized in [PEP 735](https://peps.python.org/pep-0735/). The table allows for declaration of optional dependency groups that are not published as part of the package metadata, unlike `[project.optional-dependencies]`. There are new `--group`, `--only-group`, and `--no-group` options throughout the uv interface.
```

Comment:
"of optional dependency groups" -- should we avoid "optional"?

Reply:
I felt it was important here and okay since I am comparing to `project.optional-dependencies` in the same sentence.